### DISCUSSION TITLE
Any possible way to add a resume function?

### DISCUSSION THREAD

**Discussion by Androxilogin** [hippie68/ftpdump#3] [Ideas]
I've tried to dump a game three different times and had some disconnects occur before reaching the end so I've had to start over from scratch each time. Last night I left the connection going for 10 hours to wake up to another crash at 75%. This tool would be extremely useful if you could come back to it, maybe set a resume flag to check files or whatnot. I'm sure this has probably already been an intention somewhere along the way and maybe impossible for one reason or another but being that there isn't a lot of discussion I figure why not throw it out there. I can see smaller games not really being an issue and I really don't have the option for a LAN connection with my current set up.

**Comment by hippie68** [comment]
It may be enough to let the script keep trying to connect infinitely. I have added the options --keep-trying and also --resume in this build:
[ftpdump.zip](https://github.com/hippie68/ftpdump/files/7654528/ftpdump.zip)

--resume is not 100% safe and could in rare scenarios corrupt files. Plus it is not safe to use a different FTP server when attempting to resume, as the servers may have different decryption methods.

I never had disconnects, so I don't know how that would look like and if the current changes will help. It would help a lot if you could run the script with options --debug and --verbose and tell me the output when it crashes/disconnects.

**Reply by Androxilogin** [reply]
Thanks for this and specifically the quick reply! I've been marveling in your work over the past week after a user on Reddit suggested I try this method for the auto decrypt function and I have to say, you're doing an awesome job. I began using your host since then and it seems to work best. I'll do a test run with this. Thanks again!
